git-branchless-record/src/lib.rs

```diff
@@ -199,7 +202,23 @@ fn record(
                             &branch_name,
                             &parent_commit.get_oid().to_string(),
                         ],
-                    )
+                    );
+                    if stash {
+                        try_exit_code!(check_out_commit(
+                            effects,
+                            git_run_info,
+                            &repo,
+                            &event_log_db,
+                            event_tx_id,
+                            Some(CheckoutTarget::Reference(reference_name.clone())),
```

Comment:
Now I've modified it to not make the early return, but I cannot picture how a `--detach --insert` or a `--stash --insert` would work. If you happen to know, could you please explain it?

Reply:
I think it's fine to not support these for now, but I think there are reasonable semantics:

- If you are on e.g. `main` and want to create a new commit without committing to `main`, then you could use `--detach` (or `--create` with a new branch name). If you want to rebase all of your other draft work based on main onto this branch, then you could pass `--insert` as well to rebase all descendants of `main` onto the new commit. I sometimes do this if there's a build or test fix that I need to use in all of my local stacks.
- I don't imagine `--stash --insert` would be that common, but I guess if you make a change and you want to apply it to already-stashed work (which are already descendants of the current commit), then you could do `--stash --insert`. It would oftentimes be equivalent to `--stash` and then `move` it on top of the already-stashed work.

Reply:
Got it! Thanks for the explanation.